### ISSUE TITLE
🚀 Feature: Integrate ace editor for templates

### ISSUE THREAD

**Issue by SebastianStehle** [novuhq/novu#2730]
### 🔖 Feature description

The variables at the right side are okay, but it would be better to have a code editor with intellisense / autocompletion and proper syntax highlighting.

### 🎤 Why is this feature needed ?

Find errors in templates before you save them.

### ✌️ How do you aim to achieve this?

1. Add ace editor.
2. Create a model of all variables
3. Provide intellisense.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by scopsy** [comment]
Thank you for the feedback! 

Do you refer to a particular channel? The E-mail editor & In-app channel on the new version use the react-ace editor already. Or you refer to the other channels here?

**Comment by SebastianStehle** [comment]
Yes, it was not obvious. I was talking about every place where you can use liquid (or whatever template syntax you use), .e.g

![image](https://user-images.githubusercontent.com/1236435/218318286-fd03c18a-3ae3-4eb7-8a44-35e340a679e3.png)

You can also use ace editor as a single line control, so it would also work for subject and so on.

**Comment by scopsy** [comment]
Didn't know this was possible with them, not sure how it will work with custom component such as buttons which are HTML generated, but we can try :)

**Comment by oba2311** [comment]
Seems like a nice upgrade - would be good to have a community member on it.

**Comment by scopsy** [comment]
I'm not sure if that's straightforward @oba2311. This will require some research + potentially updating the design depending on the capabilities of ACE.

**Comment by oba2311** [comment]
So I got it wrong - I  thought this is changing the under-the-hood completion mechanism?

**Comment by scopsy** [comment]
@oba2311 no, it is actually to add the completion mechanism for all places where we allow to edit content similarly to the E-mail custom code functionality

**Comment by oba2311** [comment]
Ok @scopsy sounds to me like not a good community ticket.

This goes to our backlog with low prio.

**Comment by scopsy** [comment]
@oba2311 we would need to have design for this, and alter some of the behaviour of the other channels. I think we should tackle this one internally.
